publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/create/CreatePublishedPublicationHandler.java

```diff
@@ -32,22 +44,34 @@ protected PublicationResponse processInput(EventReference eventDetail,
 
         return attempt(() -> parseInput(input))
             .map(CreatePublicationRequest::toPublication)
-            .map(this::addArbitraryIdentifierToAvoidExceptionWhenGeneratingResponse)
+            .map(this::addOwnerAndPublisher)
+            .map(this::storePublicationAsPublishedPublication)
             .map(PublicationResponse::fromPublication)
             .orElseThrow();
     }
 
+    private Publication addOwnerAndPublisher(Publication publication) {
+        Organization customer = new Builder().withId(UNIT_CUSTOMER_ID).build();
+        ResourceOwner resourceOwner = new ResourceOwner(randomUnitUser(), HARDCODED_OWNER_AFFILIATION);
+        return publication.copy().withPublisher(customer).withResourceOwner(resourceOwner).build();
+    }
+
+    private String randomUnitUser() {
+        return randomString() + "@unit.no";
+    }
+
+    private Publication storePublicationAsPublishedPublication(Publication pub) throws TransactionFailedException {
```

Comment:
probably storeAsPublication since the input is clear from the signature, also publication, not pub